client/src/modules/list/components/ListHeader.jsx

```diff
@@ -0,0 +1,237 @@
+import React, { Fragment, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+
+import { ListIcon } from 'assets/images/icons';
+import { updateList } from 'modules/list/model/actions';
+import { RouterMatchPropType } from 'common/constants/propTypes';
+import { whiteSpaceOnly } from 'common/utils/helpers';
+
+class ListHeader extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    const { details } = this.props;
+    const { description, name } = details;
+
+    this.state = {
+      description,
+      isDescriptionTextareaVisible: false,
+      isNameInputVisible: false,
+      name
+    };
+
+    this.descriptionTextarea = React.createRef();
+    this.nameInput = React.createRef();
+  }
+
+  componentDidMount() {
+    document.addEventListener('keydown', this.handleKeyPress);
+    document.addEventListener('mousedown', this.handleClick);
+  }
+
+  componentDidUpdate() {
+    this.descriptionTextarea.current &&
+      this.descriptionTextarea.current.focus();
+    this.nameInput.current && this.nameInput.current.focus();
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.handleKeyPress);
+    document.removeEventListener('mousedown', this.handleClick);
+  }
+
+  handleNameInputVisibility = () =>
+    this.setState(({ isNameInputVisible }) => ({
+      isNameInputVisible: !isNameInputVisible
+    }));
+
+  handleDescriptionTextareaVisibility = () =>
+    this.setState(({ isDescriptionTextareaVisible }) => ({
+      isDescriptionTextareaVisible: !isDescriptionTextareaVisible
+    }));
+
+  handleNameChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.setState({ name: value });
```

Comment:
trim the name here, before you save it in state, and remove trimming from render method

Reply:
what about spaces in names? 
If I want to set ` My Cohort `. 
If I trim it here I can only get `MyCohort`

Reply:
The trim() method removes whitespace from both sides of a string.
[trim](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim)

Reply:
Exactly :) 
When writing `Name<space>` space is at the end so I can't perform adding space and the second word.

I got NameUpdated instead of Name Updated.